pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -32,11 +32,11 @@ def verify_loop_configurations(trainer: "pl.Trainer", model: "pl.LightningModule
         __verify_train_val_loop_configuration(trainer, model)
         __verify_manual_optimization_support(trainer, model)
         __check_training_step_requires_dataloader_iter(model)
-    elif trainer.state.fn == TrainerFn.VALIDATING:
+    elif trainer.state.fn is TrainerFn.VALIDATING:
```

Comment:
Even though Python recommends `is` to check Enums, since we use our own weird enums which act as strings, `==` is preferred over `is` because strings shouldn't be compared with the latter.

To fix the `mypy` error, we should instead

```
if trainer.state.fn is None:
    raise ValueError("Unexpected, should be set")
```

Reply:
I checked already, it works fine. You can create a TrainerFn from a string and then compare it with "is". 

```py
from pytorch_lightning.trainer.states import TrainerFn
state = TrainerFn.FITTING
state is TrainerFn.FITTING
Out[4]: True
state = TrainerFn("fit")
state is TrainerFn.FITTING
Out[8]: True
state is TrainerFn.VALIDATING
Out[9]: False
```

Reply:
That assumes that `trainer.state.fn` will be a `TrainerFn` enum instance, but since we don't want the users to deal with the Enums, it is possible for an user to do:

```py
trainer.state.fn = "fit"
```

and then this would happen

```python
"fit" is TrainerFn.FITTING 
<string>:1: SyntaxWarning: "is" with a literal. Did you mean "=="?
Out[6]: False
```